internal/pkg/skuba/upgrade/addon/versions_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package addon
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/version"
+	clientSet "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
+	skubaconfig "github.com/SUSE/skuba/internal/pkg/skuba/skuba"
+)
+
+func TestUpdatedAddons(t *testing.T) {
+	tests := []struct {
+		name           string
+		clusterVersion *version.Version
+		clientSet      clientSet.Interface
+		expected       AddonVersionInfoUpdate
+		expectedErr    bool
+	}{
+		{
+			name:           "no skuba-config ConfigMap",
+			clusterVersion: version.MustParseGeneric("1.15.1"),
+			clientSet:      fake.NewSimpleClientset(),
+			expected: AddonVersionInfoUpdate{
+				Current: kubernetes.AddonsVersion{},
+				Updated: kubernetes.AddonsVersion{},
+			},
+		},
+		{
+			name:           "kubernetes version 1.15.2",
+			clusterVersion: version.MustParseGeneric("1.15.2"),
+			clientSet: fake.NewSimpleClientset(&corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      skubaconfig.ConfigMapName,
+					Namespace: metav1.NamespaceSystem,
+				},
+			}),
+			expected: AddonVersionInfoUpdate{
+				Current: kubernetes.AddonsVersion{
+					kubernetes.Cilium:  nil,
+					kubernetes.Kured:   nil,
+					kubernetes.Dex:     nil,
+					kubernetes.Gangway: nil,
+					kubernetes.PSP:     nil,
+				},
+				Updated: kubernetes.AddonsVersion{
+					kubernetes.Cilium:  &kubernetes.AddonVersion{Version: "1.5.3", ManifestVersion: 0},
+					kubernetes.Kured:   &kubernetes.AddonVersion{Version: "1.2.0", ManifestVersion: 0},
+					kubernetes.Dex:     &kubernetes.AddonVersion{Version: "2.16.0", ManifestVersion: 1},
+					kubernetes.Gangway: &kubernetes.AddonVersion{Version: "3.1.0-rev4", ManifestVersion: 2},
+					kubernetes.PSP:     &kubernetes.AddonVersion{Version: "1.0.0", ManifestVersion: 0},
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			got, gotErr := UpdatedAddons(tt.clientSet, tt.clusterVersion)
```

Comment:
Ideally we can have helper functions so if we remove versions in the future from the map these tests don't need any adaptation.

Have a look at https://github.com/suse/skuba/blob/c7e9c95c047aa6915822f28a8a53df941af1a43c/internal/pkg/skuba/kubernetes/versions_test.go#L27-L67 on how we achieved this by using helper functions and some refactors; the tests could provide the available versions so we test the logic strictly without relying on the real `Versions` map.